From 0ab51df991e2ede4f207e4f9b75db8294e8b2ba5 Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Thu, 8 Feb 2018 16:49:38 +0000
Subject: [PATCH 501/502] zero extra registers

This for Zero used caller-saved general registers upon function return.
---
 arch/x86/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index 8f6e7eb8ae9f..8262ba9a22dd 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -224,7 +224,7 @@ KBUILD_CFLAGS += -fno-asynchronous-unwind-tables
 # Avoid indirect branches in kernel to deal with Spectre
 ifdef CONFIG_RETPOLINE
 ifneq ($(RETPOLINE_CFLAGS),)
-  KBUILD_CFLAGS += $(RETPOLINE_CFLAGS) -DRETPOLINE
+  KBUILD_CFLAGS += $(RETPOLINE_CFLAGS) -DRETPOLINE -mzero-caller-saved-regs=used
 endif
 endif
 
-- 
2.19.1

